components/FA1_2.tsx

```diff
@@ -0,0 +1,203 @@
+import { Field, useFormikContext } from "formik";
+import { useCallback, useContext, useEffect, useRef, useState } from "react";
+import { API_URL, THUMBNAIL_URL } from "../context/config";
+import { AppStateContext } from "../context/state";
+import { debounce } from "../utils/timeout";
+import { proposals } from "../versioned/interface";
+import Alias from "./Alias";
+import ErrorMessage from "./ErrorMessage";
+import RenderTokenOption from "./RenderTokenOption";
+import Select from "./Select";
+import renderError from "./formUtils";
+
+type props = {
+  index: number;
+  children: (token: fa1_2Token | undefined) => React.ReactNode;
+  remove: (index: number) => void;
+};
+
+export type fa1_2Token = {
+  id: number;
+  account: {
+    address: string;
+  };
+  token: {
+    id: number;
+    contract: {
+      address: string;
+    };
+    tokenId: "0";
+    standard: "fa1.2";
+    totalSupply: string;
+    metadata: {
+      name: string;
+      symbol: string;
+      decimals: string;
+      thumbnailUri: string;
+    };
+  };
+  balance: string;
+  transfersCount: number;
+  firstLevel: number;
+  firstTime: string;
+  lastLevel: number;
+  lastTime: string;
+};
+
+type option = {
+  id: string;
+  value: string;
+  label: string;
+  tokenId: string;
+  image: string | undefined;
+  contractAddress: string;
+  token: fa1_2Token;
+};
+
+const FETCH_COUNT = 20;
+
+const tokenToOption = (fa1_2Token: fa1_2Token) => {
+  const { token } = fa1_2Token;
+  const imageUri = token.metadata.thumbnailUri ?? "";
+
+  return {
+    id: token.id.toString(),
+    tokenId: token.tokenId,
+    value: token.id.toString(),
+    label: token.metadata.name,
+    image: imageUri.includes("http")
+      ? imageUri
+      : imageUri === ""
+      ? undefined
+      : `${THUMBNAIL_URL}/${imageUri.replace("ipfs://", "")}`,
+    contractAddress: token.contract.address,
+    token: fa1_2Token,
+  };
+};
+
+const FA1_2 = ({ index, remove, children }: props) => {
+  const state = useContext(AppStateContext)!;
+  const { setFieldValue, getFieldProps, errors } =
+    useFormikContext<proposals>();
+
+  const [isFetching, setIsFetching] = useState(true);
+  const [canSeeMore, setCanSeeMore] = useState(true);
+
+  const [filterValue, setFilterValue] = useState<string>("");
+  const [currentToken, setCurrentToken] = useState<fa1_2Token | undefined>();
+  const [options, setOptions] = useState<option[]>([]);
+  const fetchOffsetRef = useRef(0);
+
+  const makeName = (key: string) => `transfers.${index}.values.${key}`;
+
+  const updateValues = (newToken: fa1_2Token) => {
+    setCurrentToken(newToken);
+    setFieldValue(makeName("token"), newToken ?? "");
+    setFieldValue(makeName("tokenId"), newToken?.token.tokenId ?? "");
+    setFieldValue(
+      makeName("fa1_2Address"),
+      newToken?.token.contract.address ?? ""
+    );
+  };
+
+  const fetchTokens = useCallback(
+    (value: string, offset: number) =>
+      fetch(
```

Comment:
How does the following suggestion and effort? We can do in next PR (I'd like to we start parsing lambda task ASAP.)

can you make retry (maybe 5 times) here, it the call doesn't get response in maybe 3 seconds? If no response, then show something like "server is no response, please retry later."